framework/src/media/FileOutputDataSource.cpp

```diff
@@ -79,8 +99,19 @@ bool FileOutputDataSource::open()
 
 bool FileOutputDataSource::close()
 {
+	switch (getAudioType()) {
+		case AUDIO_TYPE_WAVE:
+			unsigned int fileSize;
+			fileSize = ftell(mFp);
+			if (!utils::write_wav_header(mFp, getChannels(), getSampleRate(), getPcmFormat(), fileSize)) {
```

Comment:
If filesize used 2 or more places, yours is right. but now it used only one place, then you'd better call function directly.

Reply:
okay.